calico/base.py

```diff
@@ -64,10 +71,18 @@ def __init__(self, type_, data, timeout=-1):
 
     def __iter__(self):
         """Get components of this action as a sequence."""
-        yield self.type_.value[0]
+        yield self.type_.value[1]
         yield self.data if self.data != pexpect.EOF else "_EOF_"
         yield self.timeout
 
+    def __str__(self):
+        """Get an str which produces this action when parsed."""
+        action_type, action_data, timeout = self
+        timeout = "" if timeout == -1 else f"# timeout: {timeout}"
+        if action_data != "_EOF_":
+            action_data = dumps(action_data)
```

Comment:
I have tried ruamel.yaml.dump but it's not doing what I want while dumping strings:
>>> yaml.dump("str")
'str\n...\n'

Reply:
This is no longer a problem. I'm able to generate the yaml file with `ruamel.yaml`. I can work on it if this is OK to merge.